librz/core/linux_heap_glibc.c

```diff
@@ -126,25 +136,282 @@ static GHT GH(get_main_arena_with_symbol)(RzCore *core, RzDebugMap *map) {
 	return main_arena;
 }
 
+static ut8 *GH(get_glibc_banner)(RzCore *core, const char *section_name,
+	const char *libc_path) {
+	RzPVector *sections = NULL;
+	RzBin *bin = core->bin;
+	RzBinFile *current_bf = rz_bin_cur(bin);
+
+	void **iter;
+	ut8 *buf = NULL;
+	ut8 *buf_parse = NULL;
+	ut8 *ret_buf = NULL;
+	RzBinSection *rz_section = NULL;
+
+	RzBinOptions opt;
+	rz_bin_options_init(&opt, -1, 0, 0, false);
+	opt.obj_opts.elf_load_sections = rz_config_get_b(core->config, "elf.load.sections");
+	opt.obj_opts.elf_checks_sections = rz_config_get_b(core->config, "elf.checks.sections");
+	opt.obj_opts.elf_checks_segments = rz_config_get_b(core->config, "elf.checks.segments");
+
+	RzBinFile *libc_buf = rz_bin_open(bin, libc_path, &opt);
+	if (!libc_buf) {
+		goto cleanup;
+	}
+
+	sections = rz_bin_object_get_sections(libc_buf->o);
+	if (!sections) {
+		goto cleanup;
+	}
+
+	rz_pvector_foreach (sections, iter) {
+		rz_section = *iter;
+		if (strncmp(rz_section->name, section_name, strlen(section_name))) {
+			continue;
+		}
+		buf = calloc(rz_section->size, 1);
+		GHT read_size = rz_buf_read_at(libc_buf->buf, rz_section->paddr, buf, rz_section->size);
+		if (read_size != rz_section->size) {
+			free(buf);
+			buf = NULL;
+			goto cleanup;
+		}
+		buf_parse = (ut8 *)rz_mem_mem((const ut8 *)buf, rz_section->size, (const ut8 *)"GNU C Library", strlen("GNU C Library"));
+		ret_buf = (ut8 *)strdup((char *)buf_parse);
+		break;
+	}
+
+cleanup:
+	if (buf) {
```

Comment:
You can just write `free(buf)`, it should work with NULL values as well

Reply:
Made the change :+1: , was using this to debug